### PR TITLE
Refactor code in departure board

### PR DESCRIPTION
**Pull request by kadhikari**
Some refactoring of the code after merging the PR: https://github.com/CanalTP/navitia/pull/3160

### CODE REVIEW

File: source/time_tables/departure_boards.cpp [hunk 1]
```diff
@@ -153,66 +172,38 @@ static void render(PbCreator& pb_creator,
...
-        }
         auto pt_display_information = schedule->mutable_pt_display_informations();
-
         pb_creator.fill(route, pt_display_information, 0);
-
-        // add informations for debug. TODO : clean that
         const auto& jpc = pb_creator.data->dataRaptor->jp_container;
         pt_display_information->set_direction(
             pb_creator.data->pt_data->stop_points[jpc.get(jp_dir.jpps.back()).sp_idx.val]->stop_area->name);
-        pt_display_information->set_description(std::string("JP = ") + std::to_string(jpp_dir.jp_idx.val)
```
Comment (pbench) [inline source/time_tables/departure_boards.cpp:178]: Why does this line disappear ?

**Review by pbench — COMMENTED** [review]
Apart for the line which seems to disappear, everything looks good to me, nice refactoring :+1:

File: source/time_tables/departure_boards.cpp [hunk 2]
```diff
@@ -65,6 +65,50 @@ static bool is_terminus_for_all_stop_times(const std::vector<routing::datetime_s
     return !stop_times.empty();
 }
 
+static void fill_date_times(PbCreator& pb_creator,
+                            pbnavitia::StopSchedule* schedule,
+                            const std::pair<unsigned int, const navitia::type::StopTime*> dt_st,
+                            const boost::optional<const std::string> calendar_id) {
```
Comment (benoit-bst) [inline source/time_tables/departure_boards.cpp]:
```suggestion
                            const std::pair<unsigned int, const navitia::type::StopTime*>& dt_st,
                            const boost::optional<const std::string>& calendar_id) {
```

**Review by benoit-bst — APPROVED** [review]

File: source/time_tables/departure_boards.cpp (hunk 1 above)
Comment (kadhikari) [reply source/time_tables/departure_boards.cpp:178]: As mentioned by @pbougue this line is purely for debug purpose.

**Review by xlqian — APPROVED** [review]

**Pull request merged**